src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -4,42 +4,162 @@
+#define CHECK_WEEKEVENT_SGOROA_GKQUIET_TORCHES \
+    CHECK_WEEKEVENT(0x24, 0x10) // Talked to shrine A (any form), GK quiet, torches lit
+#define CHECK_WEEKEVENT_SGOROB_GKLOUD_NONGORON CHECK_WEEKEVENT(0x24, 0x20) // Talked to shrine B (non Goron), GK bawling
+#define CHECK_WEEKEVENT_SGOROB_GKLOUD_GORON CHECK_WEEKEVENT(0x24, 0x40)    // Talked to shrine B (as Goron), GK bawling
+#define CHECK_WEEKEVENT_SGOROB_GKQUIET_NONGORON CHECK_WEEKEVENT(0x24, 0x80) // Talked to shrine B (non Goron), GK quiet
+#define CHECK_WEEKEVENT_SGOROB_GKQUIET_GORON CHECK_WEEKEVENT(0x25, 0x01)    // Talked to shrine B (as Goron), GK quiet
+#define CHECK_WEEKEVENT_SGOROC_GKLOUD_GORON CHECK_WEEKEVENT(0x25, 0x02)     // Talked to shrine C (as Goron), GK bawling
+#define CHECK_WEEKEVENT_SGOROC_GKQUIET_GORON CHECK_WEEKEVENT(0x25, 0x04)    // Talked to shrine C (as Goron), GK quiet
+#define SET_WEEKEVENT_SGOROA_GKLOUD_NONGORON SET_WEEKEVENT(0x24, 0x01)
+#define SET_WEEKEVENT_SGOROA_GKLOUD_GORON SET_WEEKEVENT(0x24, 0x02)
+#define SET_WEEKEVENT_SGOROA_GKQUIET_NONGORON SET_WEEKEVENT(0x24, 0x04)
+#define SET_WEEKEVENT_SGOROA_GKQUIET_GORON SET_WEEKEVENT(0x24, 0x08)
+#define SET_WEEKEVENT_SGOROA_GKQUIET_TORCHES SET_WEEKEVENT(0x24, 0x10)
+#define SET_WEEKEVENT_SGOROB_GKLOUD_NONGORON SET_WEEKEVENT(0x24, 0x20)
+#define SET_WEEKEVENT_SGOROB_GKLOUD_GORON SET_WEEKEVENT(0x24, 0x40)
+#define SET_WEEKEVENT_SGOROB_GKQUIET_NONGORON SET_WEEKEVENT(0x24, 0x80)
+#define SET_WEEKEVENT_SGOROB_GKQUIET_GORON SET_WEEKEVENT(0x25, 0x01)
+#define SET_WEEKEVENT_SGOROC_GKLOUD_GORON SET_WEEKEVENT(0x25, 0x02)
+#define SET_WEEKEVENT_SGOROC_GKQUIET_GORON SET_WEEKEVENT(0x25, 0x04)
+
+#define ACTOR_USAGE_CONTEXT (this->actor.params & 0x000F)
+#define ENSGORO_INUSAGEGROUP_WSHRINE ACTOR_USAGE_CONTEXT < 3
+#define ENSGORO_USAGE_SHRINE_WINTER_A 0
+#define ENSGORO_USAGE_SHRINE_WINTER_B 1
+#define ENSGORO_USAGE_SHRINE_WINTER_C 2
+#define ENSGORO_USAGE_SHRINE_SPRING_A 3
+#define ENSGORO_USAGE_SHRINE_SPRING_B 4
+#define ENSGORO_USAGE_SHRINE_SPRING_C 5
+#define ENSGORO_USAGE_SHRINE_SPRING_D 6
+#define ENSGORO_USAGE_SHRINE_SPRING_E 7
+#define ENSGORO_USAGE_SHRINE_SPRING_F 8
+#define ENSGORO_USAGE_BOMBSHOP 9
+
+void EnSGoro_Init(Actor* thisx, PlayState* playState);
+void EnSGoro_Destroy(Actor* thisx, PlayState* playState);
+void EnSGoro_Update(Actor* thisx, PlayState* playState);
+void EnSGoro_Draw(Actor* thisx, PlayState* playState);
+
+const ActorInit En_S_Goro_InitVars = { ACTOR_EN_S_GORO,
+                                       ACTORCAT_NPC,
+                                       FLAGS,
+                                       OBJECT_OF1D_MAP,
+                                       sizeof(EnSGoro),
+                                       (ActorFunc)EnSGoro_Init,
+                                       (ActorFunc)EnSGoro_Destroy,
+                                       (ActorFunc)EnSGoro_Update,
+                                       (ActorFunc)EnSGoro_Draw };
```

Comment:
```suggestion
                                       (ActorFunc)EnSGoro_Draw, };
```
It formats better with the trailing comma.